src/jenkins.py

```diff
@@ -245,30 +250,114 @@ def _install_configs(container: ops.Container) -> None:
+        yield f"'{proxy_config.https_proxy.user or ''}'"
+        yield f"'{proxy_config.https_proxy.password or ''}'"
+    else:
+        # http proxy and https proxy value cannot both be None since proxy_config would be parsed
+        # as None.
+        proxy_config.http_proxy = typing.cast(HttpUrl, proxy_config.http_proxy)
+        yield f"'{proxy_config.http_proxy.host}'"
+        yield f"{proxy_config.http_proxy.port}"
+        yield f"'{proxy_config.http_proxy.user or ''}'"
+        yield f"'{proxy_config.http_proxy.password or ''}'"
+    if proxy_config.no_proxy:
+        yield f"'{proxy_config.no_proxy}'"
+
+
+def _configure_proxy(
+    container: ops.Container,
+    proxy_config: state.ProxyConfig | None = None,
+    client: jenkinsapi.jenkins.Jenkins | None = None,
+) -> None:
+    """Configure Jenkins proxy settings if proxy configuration values are provided.
+
+    Args:
+        container: The Jenkins workload container
+        proxy_config: The proxy settings to apply.
+        client: The API client used to communicate with the Jenkins server.
+
+    Raises:
+        JenkinsProxyError: if an error occurred running proxy configuration script.
+    """
+    if not proxy_config:
+        return
+
+    client = client if client is not None else _get_client(get_admin_credentials(container))
+    parsed_args = ", ".join(_get_groovy_proxy_args(proxy_config))
+    try:
+        client.run_groovy_script(f"proxy = new ProxyConfiguration({parsed_args})\nproxy.save()")
+    except jenkinsapi.custom_exceptions.JenkinsAPIException as exc:
+        logger.error("Failed to configure proxy, %s", exc)
+        raise JenkinsProxyError("Proxy configuration failed.") from exc
+
+
+def _get_java_proxy_args(proxy_config: state.ProxyConfig) -> typing.Iterable[str]:
+    """Get JVM system property arguments for proxy.
+
+    Args:
+        proxy_config: The proxy settings to apply.
+
+    Yields:
+        JVM System property proxy arguments.
+    """
+    if proxy_config.http_proxy:
+        yield f"-Dhttp.proxyHost={proxy_config.http_proxy.host}"
+        yield f"-Dhttp.proxyPort={proxy_config.http_proxy.port}"
+        if proxy_config.http_proxy.user and proxy_config.http_proxy.password:
+            yield f"-Dhttp.proxyUser={proxy_config.http_proxy.user}"
+            yield f"-Dhttp.proxyPassword={proxy_config.http_proxy.password}"
+    if proxy_config.https_proxy:
+        yield f"-Dhttps.proxyHost={proxy_config.https_proxy.host}"
+        yield f"-Dhttps.proxyPort={proxy_config.https_proxy.port}"
+        if proxy_config.https_proxy.user and proxy_config.https_proxy.password:
+            yield f"-Dhttps.proxyUser={proxy_config.https_proxy.user}"
+            yield f"-Dhttps.proxyPassword={proxy_config.https_proxy.password}"
+    if proxy_config.no_proxy:
+        formatted_no_proxy_hosts = "|".join(proxy_config.no_proxy.split(","))
+        yield f'-Dhttp.nonProxyHosts="{formatted_no_proxy_hosts}"'
```

Comment:
minor/can-be-addressed-later: Does it exists a https.nonProxyHosts configuration?

Reply:
Oh, the comment got removed during the refactor! I will add it back in a future push!
Http and https shares the same configuration for no proxy! :D